pipeline/metadata/flatten_satellite.py

```diff
@@ -371,54 +464,60 @@ def _process_satellite_v2_control(
         Rows
     """
     responses = scan.get('response', [])
-    if responses:
-      # An overall satellite v2 measurement
-      # always contains some non-control trial domains
-      is_control_domain = False
+
+    for response in responses:
+      is_control_domain = flatten_base.is_control_url(response['url'])
 
       row = {
           'domain':
-              scan['test_url'],
+              response['url'],
           'is_control':
               is_control_domain,
           'category':
-              self.category_matcher.get_category(scan['test_url'],
+              self.category_matcher.get_category(response['url'],
                                                  is_control_domain),
           'ip':
               scan['vp'],
           'is_control_ip':
               True,
           'date':
-              responses[0]['start_time'][:10],
+              response['start_time'][:10],
           'start_time':
-              format_timestamp(responses[0]['start_time']),
+              format_timestamp(response['start_time']),
           'end_time':
-              format_timestamp(responses[-1]['end_time']),
+              format_timestamp(response['end_time']),
           'anomaly':
               None,
           'success':
               not scan['connect_error'],
```

Comment:
I don't see `connect_error` and 'passed_control` in the `responses.json` type definition.

Reply:
This code is heavily unit tested (with the unit tests coming from example measurements) and has been run against the entire satellite dataset. If you're seeing fields that don't match the documentation it's probably that the docs are wrong.

Here's the example measurement in the unit tests with this field https://github.com/censoredplanet/censoredplanet-analysis/blob/16276c030dc9482c3c0601850f76503001c36cb4/pipeline/metadata/test_satellite.py#L560